programs/fileio.c

```diff
@@ -2169,43 +2169,33 @@ FIO_determineDstName(const char* srcFileName)
     static size_t dfnbCapacity = 0;
     static char* dstFileNameBuffer = NULL;   /* using static allocation : this function cannot be multi-threaded */
 
+    const char* SUFFIX_LIST = ZSTD_EXTENSION "/" TZSTD_EXTENSION
+    #ifdef ZSTD_GZDECOMPRESS
+        "/" GZ_EXTENSION "/" TGZ_EXTENSION
+    #endif
+    #ifdef ZSTD_LZMADECOMPRESS
+        "/" XZ_EXTENSION "/" LZMA_EXTENSION "/" TXZ_EXTENSION
+    #endif
+    #ifdef ZSTD_LZ4DECOMPRESS
+        "/" LZ4_EXTENSION "/" TLZ4_EXTENSION
+    #endif
+    ;
+
     size_t const sfnSize = strlen(srcFileName);
     size_t suffixSize;
     const char* const suffixPtr = strrchr(srcFileName, '.');
     if (suffixPtr == NULL) {
-        DISPLAYLEVEL(1, "zstd: %s: unknown suffix -- ignored \n",
-                        srcFileName);
+        DISPLAYLEVEL(1, "zstd: %s: missing suffix (%s expected). Can't derive the output file name so specify it with -o dstFileName. -- ignored \n",
+                        srcFileName, SUFFIX_LIST);
         return NULL;
     }
     suffixSize = strlen(suffixPtr);
 
     /* check suffix is authorized */
     if (sfnSize <= suffixSize
-        || (   strcmp(suffixPtr, ZSTD_EXTENSION)
-        #ifdef ZSTD_GZDECOMPRESS
-            && strcmp(suffixPtr, GZ_EXTENSION)
-        #endif
-        #ifdef ZSTD_LZMADECOMPRESS
-            && strcmp(suffixPtr, XZ_EXTENSION)
-            && strcmp(suffixPtr, LZMA_EXTENSION)
-        #endif
-        #ifdef ZSTD_LZ4DECOMPRESS
-            && strcmp(suffixPtr, LZ4_EXTENSION)
-        #endif
-            ) ) {
-        const char* suffixlist = ZSTD_EXTENSION
-        #ifdef ZSTD_GZDECOMPRESS
-            "/" GZ_EXTENSION
-        #endif
-        #ifdef ZSTD_LZMADECOMPRESS
-            "/" XZ_EXTENSION "/" LZMA_EXTENSION
-        #endif
-        #ifdef ZSTD_LZ4DECOMPRESS
-            "/" LZ4_EXTENSION
-        #endif
-        ;
-        DISPLAYLEVEL(1, "zstd: %s: unknown suffix (%s expected) -- ignored \n",
-                     srcFileName, suffixlist);
+        || (strstr(SUFFIX_LIST, suffixPtr) == NULL)) {
```

Comment:
Won't this also accept a suffix like `.z` (i.e., a prefix of a valid extension)?